src/main/java/org/zeplinko/commons/lang/ext/core/Retryable.java

```diff
@@ -0,0 +1,225 @@
+package org.zeplinko.commons.lang.ext.core;
+
+import jakarta.annotation.Nonnull;
+
+import java.util.Objects;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+
+/**
+ * Executes a {@link Callable} with retry semantics.
+ * <p>
+ * A {@code Retryable} encapsulates a task that may be re-executed in the event
+ * of failure or validation rejection. Clients configure the behaviour fluently
+ * using:
+ * </p>
+ * <ul>
+ * <li>{@link #retryOnResult(Predicate)} – validation predicate that must hold
+ * for the <em>result</em> to be accepted</li>
+ * <li>{@link #retryOnFailure(Predicate)} – declare <em>exceptions</em> that
+ * trigger a retry</li>
+ * <li>{@link #retries(int)} – maximum retry attempts</li>
+ * <li>{@link #delay(long, TimeUnit)} – base delay between attempts</li>
+ * <li>{@link #backoff(Backoff)} – back-off strategy for calculating the next
+ * delay</li>
+ * </ul>
+ * The task is started by calling {@link #run()} which returns a {@link Result}
+ * describing the outcome.
+ *
+ * @param <R> the type of the successful result returned by the task
+ */
+public final class Retryable<R> {
+    private static final int MAX_RETRIES_UPPER_BOUND = 1000;
+
+    private final Callable<R> task;
+
+    private Predicate<? super R> resultPredicate = r -> true; // Validation predicate: accept any result by default
+
+    private Predicate<? super Exception> exceptionPredicate = e -> false; // Do not retry on exceptions by default
+
+    private int maxRetries = 0;
```

Comment:
I understand the rationale to keep this as 0 to ensure that the retries doesn't happen unless explicitly defined.
But if want it to be similar to Spring Retryable, we should keep the maxRetries as 3.
[Spring Doc](https://docs.spring.io/spring-retry/docs/current/apidocs/org/springframework/retry/annotation/Retryable.html#maxAttempts())